metrics/core/src/main/java/org/hyperledger/besu/metrics/opentelemetry/OpenTelemetrySystem.java

```diff
@@ -0,0 +1,295 @@
+import org.hyperledger.besu.plugin.services.metrics.LabelledMetric;
+import org.hyperledger.besu.plugin.services.metrics.MetricCategory;
+import org.hyperledger.besu.plugin.services.metrics.OperationTimer;
+
+import java.lang.management.GarbageCollectorMXBean;
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryMXBean;
+import java.lang.management.MemoryPoolMXBean;
+import java.lang.management.MemoryUsage;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.DoubleSupplier;
+import java.util.stream.Stream;
+
+import com.google.common.collect.ImmutableSet;
+import io.opentelemetry.common.Labels;
+import io.opentelemetry.exporters.otlp.OtlpGrpcMetricExporter;
+import io.opentelemetry.metrics.DoubleValueObserver;
+import io.opentelemetry.metrics.DoubleValueRecorder;
+import io.opentelemetry.metrics.LongCounter;
+import io.opentelemetry.metrics.LongSumObserver;
+import io.opentelemetry.metrics.LongUpDownSumObserver;
+import io.opentelemetry.metrics.Meter;
+import io.opentelemetry.sdk.metrics.MeterSdkProvider;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.export.IntervalMetricReader;
+
+/** Metrics system relying on the native OpenTelemetry format. */
+public class OpenTelemetrySystem implements ObservableMetricsSystem {
+  private static final String TYPE_LABEL_KEY = "type";
+  private static final String AREA_LABEL_KEY = "area";
+  private static final String POOL_LABEL_KEY = "pool";
+  private static final String USED = "used";
+  private static final String COMMITTED = "committed";
+  private static final String MAX = "max";
+  private static final String HEAP = "heap";
+  private static final String NON_HEAP = "non_heap";
+
+  private final Set<MetricCategory> enabledCategories;
+  private final boolean timersEnabled;
+  private final Map<String, LabelledMetric<Counter>> cachedCounters = new ConcurrentHashMap<>();
+  private final Map<String, LabelledMetric<OperationTimer>> cachedTimers =
+      new ConcurrentHashMap<>();
+  private final MeterSdkProvider meterSdkProvider = MeterSdkProvider.builder().build();
+  private IntervalMetricReader periodicReader;
+
+  public OpenTelemetrySystem(
+      final Set<MetricCategory> enabledCategories,
+      final boolean timersEnabled,
+      final boolean pushEnabled) {
+    this.enabledCategories = ImmutableSet.copyOf(enabledCategories);
+    this.timersEnabled = timersEnabled;
+
+    if (pushEnabled) {
+      IntervalMetricReader.Builder builder =
+          IntervalMetricReader.builder()
+              .setMetricProducers(Collections.singleton(meterSdkProvider.getMetricProducer()))
+              .setMetricExporter(OtlpGrpcMetricExporter.getDefault())
+              .setExportIntervalMillis(500);
```

Comment:
Export at 2Hz?  Is that a good idea performance wise?  I think you should use the metrics push interval from the MetricsConfiguration.  If we need to go faster than 1/sec we may need to adjust it somehow.  But I would need to be persuaded that pushing more than 1/sec is a good idea at scale.

Reply:
Right, sounds like a good config flag. I know Prometheus is routinely 5 to 10s. Otel can be faster, especially if you want to instrument realtime metrics. A good default is 5s.

Reply:
if we need to we can add a `--push-metrics-interval-unit` that defaults to seconds, but if Otel doesn't need sub-second push frequency I'd prefer to push it off as long as possible.

Reply:
It's not about Otel needing sub-second frequency, it's what the users want to see. Some folks use Otel and backend offering for subsecond metric resolution. Granted, that's probably not a prime use case here, so we can stick to seconds until someone comes along with that need.